api/v1alpha1/eso_types.go

```diff
@@ -0,0 +1,114 @@
+type ExternalSecretsOperatorList struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard list's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ListMeta `json:"metadata"`
+	Items           []ExternalSecretsOperator `json:"items"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+
+// ExternalSecretsOperator describes configuration and information about the managed external-secrets
+// deployment. The name must be `cluster` to make ExternalSecretsOperator a singleton that is, to
+// allow only one instance of ExternalSecretsOperator per cluster.
+//
+// ExternalSecretsOperator is mainly for configuring the global options and enabling the features, which
+// serves as a common/centralized config for managing multiple controllers of the operator. The object
+// will be created during the operator installation.
+//
+// +kubebuilder:validation:XValidation:rule="self.metadata.name == 'cluster'",message="ExternalSecretsOperator is a singleton, .metadata.name must be 'cluster'"
+// +operator-sdk:csv:customresourcedefinitions:displayName="ExternalSecretsOperator"
+type ExternalSecretsOperator struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is the specification of the desired behavior of the ExternalSecretsOperator.
+	// +kubebuilder:validation:Required
+	Spec ExternalSecretsOperatorSpec `json:"spec,omitempty"`
+
+	// status is the most recently observed status of the ExternalSecretsOperator.
+	Status ExternalSecretsOperatorStatus `json:"status,omitempty"`
+}
+
+// ExternalSecretsOperatorSpec is the specification of the desired behavior of the ExternalSecretsOperator.
+type ExternalSecretsOperatorSpec struct {
+	// globalConfig is for configuring the external-secrets-operator behavior.
+	// +kubebuilder:validation:Optional
+	GlobalConfig *GlobalConfig `json:"globalConfig,omitempty"`
+
+	// features is for enabling the optional features.
+	Features []Feature `json:"features,omitempty"`
+}
+
+// GlobalConfig is for configuring the external-secrets-operator behavior.
+type GlobalConfig struct {
+	// logLevel supports value range as per [kubernetes logging guidelines](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md#what-method-to-use).
+	// +kubebuilder:default:=1
+	// +kubebuilder:validation:Minimum:=1
+	// +kubebuilder:validation:Maximum:=5
+	// +kubebuilder:validation:Optional
+	LogLevel int32 `json:"logLevel,omitempty"`
+
+	// resources is for defining the resource requirements.
+	// Cannot be updated.
```

Comment:
```
	// Cannot be updated.
```
What would it mean?

Reply:
It's an immutable field, once configured it cannot be updated.
Ref: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#podspec-v1-core:~:text=Compute%20Resources%20required%20by%20this%20container.%20Cannot%20be%20updated.

I think we can add an CEL validation for this too.